yarn-project/sequencer-client/src/publisher/sequencer-publisher.ts

```diff
@@ -354,20 +376,25 @@ export class SequencerPublisher {
 
     const ts = BigInt((await this.l1TxUtils.getBlock()).timestamp + this.ethereumSlotDuration);
 
-    // use sender balance to simulate
-    const balance = await this.l1TxUtils.getSenderBalance();
-    this.log.debug(`Simulating validateHeader with balance: ${balance}`);
+    // In fisherman mode, we can't know where the proposer is publishing from
+    // so we simulate as the proposer but with sufficient balance
+    const stateOverrides = await this.rollupContract.makePendingBlockNumberOverride(opts?.forcePendingBlockNumber);
+    if (this.proposerAddressForSimulation) {
+      stateOverrides.push({
+        address: this.proposerAddressForSimulation.toString(),
+        balance: 10n * WEI_CONST * WEI_CONST, // 10 ETH
+      });
+    }
+
     await this.l1TxUtils.simulate(
       {
         to: this.rollupContract.address,
         data: encodeFunctionData({ abi: RollupAbi, functionName: 'validateHeaderWithAttestations', args }),
-        from: MULTI_CALL_3_ADDRESS,
+        // In fisherman mode, simulate as if the proposer is sending the tx
+        ...(this.proposerAddressForSimulation && { from: this.proposerAddressForSimulation.toString() }),
       },
       { time: ts + 1n },
-      [
-        { address: MULTI_CALL_3_ADDRESS, balance },
-        ...(await this.rollupContract.makePendingBlockNumberOverride(opts?.forcePendingBlockNumber)),
-      ],
+      stateOverrides,
```

Comment:
Heads up we're not sending the proposal from the multicall anymore here, not setting its balance. Not sure if it affects things though, where does the simulation run from if no `from` is set?